README.md

```diff
@@ -14,628 +14,1726 @@ Namespace: `api.common` in [Impress Application Server](https://github.com/metar
+Split array into two parts
 
-### Clone Object
-`common.clone(obj)`
-- `obj:object or array`
 
-Returns: object or array
+#### shuffle(arr)
 
-### Duplicate object ot array
-`common.duplucate(obj)`
-- `obj:object or array`
+- `arr`: [`<Array>`]
 
-Returns: object or array
+_Returns:_ [`<Array>`]
 
-### Read property by dot-separated path
-`common.getByPath(data, dataPath)`
-- `data:hash`
-- `dataPath:string` - dot-separated path
+Shuffle an array
 
-Returns: value
 
-### Set property by dot-separated path
-`common.setByPath(data, dataPath, value)`
-- `data:hash`
-- `dataPath:string` - dot-separated path
-- `value` - new value
+#### range(from, to)
 
-### Delete property by dot-separated path
-`common.deleteByPath(data, dataPath)`
-- `data:object`
-- `dataPath:string` - dot-separated path
+- `from`: [`<number>`] range start
+- `to`: [`<number>`] range end
 
-Returns: boolean
+_Returns:_ [`<Array>`]
 
-### Distinct merge multiple arrays
-`common.merge(...args)`
-- `args:array of array`
+Generate int array from given range
 
-Returns: array
+_Example:_
+```js
+ range(1, 5)
+```
+
+_Result:_
+```js
+ [1, 2, 3, 4, 5]
+```
+
+
+#### sequence(seq, max)
+
+- `seq`: [`<Array>`]
+- `max`: [`<number>`] (optional) max
+
+_Returns:_ [`<Array>`]
+
+Generate int array from sequence syntax
+
+_Example:_
+```js
+ list: sequence([81, 82, 83])
+```
+
+_Result:_
+```js
+ [81, 82, 83]
+```
+
+_Example:_
+```js
+ range from..to: sequence([81,,83]) = [81, 82, 83]
+```
+
+_Result:_
+```js
+ [81, 82, 83]
+```
+
+_Example:_
+```js
+ range from..count: sequence([81, [3]]) = [81, 82, 83]
+```
+
+_Result:_
+```js
+ [81, 82, 83]
+```
+
+_Example:_
+```js
+ range from..max-to: sequence([81, [-2]], 5) = [81, 82, 83]
+```
+
+_Result:_
+```js
+ [81, 82, 83]
+```
+
+
+#### last(arr)
+
+- `arr`: [`<Array>`]
+
+_Returns:_ `<any>` element
+
+Get last element of array
+
+
+#### checkLogin(login, required, optional)
+
+- `login`: [`<string>`] login to test
+- `required`: [`<Array>`] required tests configs
+- `optional`: [`<Array>`] optional tests configs, defalult: `[]`
+
+_Returns:_ `<AuthenticationStrength>`
+
+Function that tests the login
+
+
+#### checkPassword(password, required, optional)
+
+- `password`: [`<string>`] password to test
+- `required`: [`<Array>`] required tests configs
+- `optional`: [`<Array>`] optional tests configs, default: `[]`
+
+_Returns:_ `<AuthenticationStrength>`
+
+Function that tests the password
+
+
+#### checkLoginPassword(login, password, required, optional)
+
+- `login`: [`<string>`] login to test
+- `password`: [`<string>`] password to test
+- `required`: [`<Array>`] required tests configs
+- `optional`: [`<Array>`] optional tests configs, default: `[]`
+
+_Returns:_ `<AuthenticationStrength>`
+
+Function that tests the login with password
+
+
+#### BTree()
+
+
+
+#### BTree.prototype.constructor()
+
+
+
+#### BTree.prototype.get()
+
+
+
+#### BTree.prototype.set()
+
+
+
+#### BTree.prototype.iterator()
+
+
+
+#### BTree.prototype.remove()
+
+
```

Comment:
I assume this is an empty line for missing arguments. I think we can omit it in this case.

/cc @nechaido @belochub @tshemsedinov